src/driver/db.ts

```diff
@@ -0,0 +1,190 @@
+    CollectionOptions,
+    CreateTableDefinition,
+    ListCollectionsOptions,
+    ListTablesOptions,
+    RawDataAPIResponse,
+    Table as AstraTable,
+    TableOptions,
+    Collection
+} from '@datastax/astra-db-ts';
+import { StargateMongooseError } from '../stargateMongooseError';
+
+/**
+ * Defines the base database class for interacting with Astra DB. Responsible for creating collections and tables.
+ * This class abstracts the operations for both collections mode and tables mode. There is a separate TablesDb class
+ * for tables and CollectionsDb class for collections.
+ */
+export abstract class BaseDb {
+    astraDb: AstraDb;
+    /**
+     * Whether we're using "tables mode" or "collections mode". If tables mode, then `collection()` returns
+     * a Table instance, **not** a Collection instance. Also, if tables mode, `createCollection()` throws an
+     * error for Mongoose `syncIndexes()` compatibility reasons.
+     */
+    useTables: boolean;
+
+    constructor(astraDb: AstraDb, keyspaceName: string, useTables?: boolean) {
+        this.astraDb = astraDb;
+        astraDb.useKeyspace(keyspaceName);
+        this.useTables = !!useTables;
+    }
+
+    /**
+     * Return the raw HTTP client used by astra-db-ts to talk to the db.
+     */
+    get httpClient() {
+        return this.astraDb._httpClient;
+    }
+
+    /**
+     * Get a collection by name.
+     * @param name The name of the collection.
+     */
+    abstract collection<DocType extends Record<string, unknown> = Record<string, unknown>>(name: string, options: Record<string, unknown>): AstraCollection | AstraTable<DocType>;
+
+    /**
+     * Create a new collection with the specified name and options.
+     * @param name The name of the collection to be created.
+     * @param options Additional options for creating the collection.
+     */
+    abstract createCollection(name: string, options?: Record<string, unknown>): Promise<Collection>;
+
+    /**
+     * Create a new table with the specified name and definition
+     * @param name
+     * @param definition
+     */
+
+    async createTable(name: string, definition: CreateTableDefinition) {
+        return this.astraDb.createTable(name, { definition });
+    }
+
+    /**
+     * Drop a collection by name.
+     * @param name The name of the collection to be dropped.
+     */
+    async dropCollection(name: string, options?: { timeout?: number }) {
+        return this.astraDb.dropCollection(name, options);
+    }
+
+    /**
+     * Drop a table by name. This function does **not** throw an error if the table does not exist.
+     * @param name
+     */
+    async dropTable(name: string) {
+        return this.astraDb.dropTable(name).catch(err => {
+            // For consistency with Mongoose, ignore errors if the table does not exist.
+            if (err.errorDescriptors?.[0].errorCode === 'CANNOT_DROP_UNKNOWN_TABLE') {
+                return;
+            }
+            throw err;
+        });
```

Comment:
```suggestion
        return this.astraDb.dropTable(name, {
            ifExists: true,
        });
```
You're probably looking for this option 🙂